File: app/src/main/java/com/davismiyashiro/expenses/injection/ApplicationModule.java
package com.davismiyashiro.expenses.injection;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @Author Davis Miyashiro
 */

@Module
public class ApplicationModule {

    private Application application;

    public ApplicationModule (Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    public Context provideContext () {
        return application;
    }
}

File: app/src/main/java/com/davismiyashiro/expenses/injection/ApplicationModule.kt
package com.davismiyashiro.expenses.injection

import android.app.Application
import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * @Author Davis Miyashiro
 */

@Module
class ApplicationModule(private val application: Application) {

    @Provides
    @Singleton
    fun provideContext(): Context {
        return application
    }
}
